### PR TITLE
Add fast_runtime! macros

### LINKED ISSUE #88: Add fast_runtime! macros supporting
The fast_runtime! macro in the Substrate, is potentially designed to enhance the performance and efficiency of blockchain runtime environments. It’s also worth looking at the mechanism for testing runtime migrations based on the state

### PR DESCRIPTION
**Pull request by DmitriVT**

### CODE REVIEW

File: runtime/src/utils.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#[macro_export]
+macro_rules! prod_or_fast {
+    ($prod:expr, $test:expr) => {
+        if cfg!(feature = "fast-runtime") {
+            $test
+        } else {
+            $prod
+        }
+    };
+    ($prod:expr, $test:expr, $env:expr) => {
```
Comment (hrls) [inline runtime/src/utils.rs]: This match arm looks like unused.

Comment (DmitriVT) [reply runtime/src/utils.rs]: Deleted

File: runtime/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/// Macro to set a value (e.g. when using the `parameter_types` macro) to either a production value
+/// or to an environment variable or testing value (in case the `fast-runtime` feature is selected).
+/// Note that the environment variable is evaluated _at compile time_.
+///
+/// Usage:
+/// ```Rust
+/// parameter_types! {
+///     // Note that the env variable version parameter cannot be const.
+///     pub LaunchPeriod: BlockNumber = prod_or_fast!(7 * DAYS, 1, "KSM_LAUNCH_PERIOD");
```
Comment (hrls) [inline runtime/src/utils.rs]: Comment lines are outdated

**Review by hrls — DISMISSED** [review]
LGTM

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -373,22 +373,22 @@ parameter_types! {
     pub const ProposalBond: Permill = Permill::from_percent(5);
     pub ProposalBondMinimum: Balance = 10 * DOLLARS;
     pub ProposalBondMaximum: Balance = 50 * DOLLARS;
-    pub const SpendPeriod: BlockNumber = 30 * DAYS;
+    pub const SpendPeriod: BlockNumber = conf!(mainnet: 30 * DAYS, testnet: 30 * DAYS);
```
Comment (ales-tsurko) [inline runtime/src/lib.rs]: why use `conf!` here if the value is the same for both variants?

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -373,22 +373,22 @@ parameter_types! {
...
     pub const Burn: Permill = Permill::from_percent(1);
 
     pub const TipCountdown: BlockNumber = 2 * DAYS;
     pub const TipFindersFee: Percent = Percent::from_percent(5);
     pub TipReportDepositBase: Balance = deposit(1, 0);
     pub BountyDepositBase: Balance = deposit(1, 0);
-    pub const BountyDepositPayoutDelay: BlockNumber = 6 * DAYS;
-    pub const BountyUpdatePeriod: BlockNumber = 35 * DAYS;
+    pub const BountyDepositPayoutDelay: BlockNumber = conf!(mainnet: 6 * DAYS, testnet: 6 * DAYS);
+    pub const BountyUpdatePeriod: BlockNumber = conf!(mainnet: 35 * DAYS, testnet: 35 * DAYS);
```
Comment (ales-tsurko) [inline runtime/src/lib.rs]: same as for the previous

Comment (ales-tsurko) [reply runtime/src/lib.rs]: and there are others below

File: runtime/src/utils.rs (hunk 2 above)
Comment (DmitriVT) [reply runtime/src/utils.rs]: Yes, agree. Updated

File: runtime/src/lib.rs (hunk 4 above)
Comment (DmitriVT) [reply runtime/src/lib.rs]: Yes, this happened due to the fact that at the moment there are questions about the configs on the mainnet. 
Updated the configs by deleting the conf! in places where it is not needed now.

**Review by hrls — DISMISSED** [review]

**Review by ales-tsurko — DISMISSED** [review]

File: runtime/src/utils.rs (hunk 2 above)
Comment (hrls) [reply runtime/src/utils.rs]: I mean arguments labels are missed too

Comment (DmitriVT) [reply runtime/src/utils.rs]: Updated

**Review by hrls — APPROVED** [review]

**Pull request merged**